gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageImpl.java

```diff
@@ -2096,9 +2115,13 @@ public GoogleCloudStorageItemInfo composeObjects(
     return compositeInfo;
   }
 
+  @VisibleForTesting
   <RequestT extends StorageRequest<?>> RequestT initializeRequest(
       RequestT request, String bucketName) throws IOException {
-    if (storageRequestAuthorizer != null) {
+    if (shouldAuthenticationPerRequest && this.accessTokenProvider != null) {
+      AccessToken accessToken = this.accessTokenProvider.getAccessToken(request);
+      request.setOauthToken(accessToken.getToken());
+    } else if (storageRequestAuthorizer != null) {
```

Comment:
so I think it is also worth mentioning at least in the CONFIGURATION.md file that "shouldAuthenticationPerRequest" will precede storage request authorizer.

Reply:
Will do. I will introduce an `AuthenticationMethod` to make it clear what is being used.

Reply:
Actually, on a second thought, I don't think we should make these two mutually exclusive. So will keep both running if they're set. I will make it clear through the CONFIGURATION.md that the access token provider is the preferred way now.